ingestion/tests/unit/profiler/pandas/test_custom_metrics.py

```diff
@@ -0,0 +1,233 @@
+            os.path.join(path, filename)
+            for path, _, files in os.walk(self.resources_dir)
+            for filename in files
+        ]
+
+        self.s3_keys = []
+
+        for path in resources_paths:
+            key = os.path.relpath(path, self.resources_dir)
+            self.s3_keys.append(key)
+            self.client.upload_file(Filename=path, Bucket=BUCKET_NAME, Key=key)
+
+        with patch.object(
+            PandasProfilerInterface,
+            "_convert_table_to_list_of_dataframe_objects",
+            return_value=self.dfs,
+        ):
+            self.sqa_profiler_interface = PandasProfilerInterface(
+                self.datalake_conn,
+                None,
+                self.table_entity,
+                None,
+                None,
+                None,
+                None,
+                None,
+                thread_count=1,
+            )
+
+    def test_table_custom_metric(self):
+        table_entity = Table(
+            id=uuid4(),
+            name="user",
+            columns=[
+                EntityColumn(
+                    name=ColumnName(__root__="id"),
+                    dataType=DataType.INT,
+                )
+            ],
+            customMetrics=[
+                CustomMetric(
+                    name="LastNameFilter",
+                    expression="'last_name' != Doe",
+                ),
+                CustomMetric(
+                    name="notUS",
+                    expression="'country == US'",
+                ),
+            ],
+        )
+        with patch.object(
+            PandasProfilerInterface,
+            "_convert_table_to_list_of_dataframe_objects",
+            return_value=self.dfs,
+        ):
+            self.sqa_profiler_interface = PandasProfilerInterface(
+                self.datalake_conn,
+                None,
+                table_entity,
+                None,
+                None,
+                None,
+                None,
+                None,
+                thread_count=1,
+            )
+
+        profiler = Profiler(
+            profiler_interface=self.sqa_profiler_interface,
+        )
+        metrics = profiler.compute_metrics()
+        for k, v in metrics._table_results.items():
+            for metric in v:
+                if metric.name == "LastNameFilter":
+                    assert metric.value == 1
+                if metric.name == "notUS":
+                    assert metric.value == 2
+
+    def test_column_custom_metric(self):
+        table_entity = Table(
+            id=uuid4(),
+            name="user",
+            columns=[
+                EntityColumn(
+                    name=ColumnName(__root__="id"),
+                    dataType=DataType.INT,
+                    customMetrics=[
+                        CustomMetric(
+                            name="LastNameFilter",
+                            columnName="id",
+                            expression="'last_name' != Doe",
```

Comment:
is the language of the expression SQL?

Reply:
it is not. it should return a boolean expression. I'll add it with the documentation https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html